itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java

```diff
@@ -319,31 +330,92 @@ public void testFailoverDuringDump() throws Throwable {
     assertTrue(fs.exists(new Path(dumpPath, ReplAck.LOAD_ACKNOWLEDGEMENT.toString())));
 
     assertTrue(MetaStoreUtils.isDbBeingFailedOver(primary.getDatabase(primaryDbName)));
-    dumpData = primary.dump(primaryDbName);
-    dumpPath = new Path(dumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR);
-    Assert.assertEquals(new DumpMetaData(dumpPath, conf).getDumpType(), DumpType.INCREMENTAL);
-    Path failoverReadyFile = new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString());
-    Path failoverMdFile = new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA);
-    assertFalse(fs.exists(failoverReadyFile));
-    assertFalse(fs.exists(failoverMdFile));
-    assertFalse(MetaStoreUtils.isDbBeingFailedOver(primary.getDatabase(primaryDbName)));
-    replica.load(replicatedDbName, primaryDbName);
 
-    fs.create(failoverReadyFile);
-    fs.create(failoverMdFile);
-    assertTrue(fs.exists(failoverReadyFile));
-    assertTrue(fs.exists(failoverMdFile));
+    primary.run("drop database if exists " + primaryDbName + " cascade");
 
-    //Since the failover start config is disabled and previous valid dump directory contains _failover_ready marker file
-    //So, this dump iteration will perform bootstrap dump instead of incremental and last dump directory also should not
-    //deleted.
-    WarehouseInstance.Tuple newDumpData = primary.dump(primaryDbName);
-    assertNotEquals(newDumpData.dumpLocation, dumpData.dumpLocation);
+    assertTrue(primary.getDatabase(primaryDbName) == null);
+
+    assertTrue(ReplChangeManager.getReplPolicyIdString(replica.getDatabase(replicatedDbName)) == null);
+    WarehouseInstance.Tuple reverseDumpData = replica.dump(replicatedDbName);
+    assertNotEquals(reverseDumpData.dumpLocation, dumpData.dumpLocation);
     assertTrue(fs.exists(dumpPath));
     assertTrue(fs.exists(new Path(dumpPath, ReplAck.FAILOVER_READY_MARKER.toString())));
-    dumpPath = new Path(newDumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR);
+    dumpPath = new Path(reverseDumpData.dumpLocation, ReplUtils.REPL_HIVE_BASE_DIR);
     assertFalse(fs.exists(new Path(dumpPath, FailoverMetaData.FAILOVER_METADATA)));
     assertTrue(new DumpMetaData(dumpPath, conf).getDumpType() == DumpType.BOOTSTRAP);
+    assertTrue(fs.exists(new Path(dumpPath, DUMP_ACKNOWLEDGEMENT.toString())));
+    assertTrue(ReplUtils.isDbBeingFailedOverAtTarget(replica.getDatabase(replicatedDbName)));
```

Comment:
Assert at this point that the _dumpmetadat has marked the dump as bootstrap one

Reply:
Already done.